assembly/src/ast/procedure/alias.rs

```diff
@@ -27,17 +28,20 @@ pub struct ProcedureAlias {
     ///
     /// NOTE: This is fully-qualified from the perspective of the containing [Module], but may not
     /// be fully-resolved to the concrete definition until compilation time.
-    pub(crate) target: FullyQualifiedProcedureName,
+    pub(crate) target: AliasTarget,
+    /// If true, this alias was created with an absolute path, bypassing the need for an import
+    absolute: bool,
 }
 
 impl ProcedureAlias {
     /// Creates a new procedure alias called `name`, which resolves to `target`.
-    pub fn new(name: ProcedureName, target: FullyQualifiedProcedureName) -> Self {
+    pub fn new(name: ProcedureName, target: impl Into<AliasTarget>, absolute: bool) -> Self {
         Self {
             docs: None,
             source_file: None,
             name,
-            target,
+            target: target.into(),
+            absolute,
         }
```

Comment:
Should there be a check that `absolute == false` is only valued for `AliasTarget::Path`? That is, digest-based aliases are always absolute.

Reply:
True, though in such cases the `absolute` flag doesn't come into play anyway, but it doesn't hurt to have an assertion here to check that the semantics of what is being defined are consistent.